smaily_for_drupal/src/Plugin/Block/SmailyForDrupalFormBlock.php

```diff
@@ -159,25 +160,27 @@ public function fetchAutoresponders() {
     $domain = $config->get('smaily_api_credentials.domain');
 
     $autoresponder_list = [];
-    if (!empty($domain) && !empty($username) && !empty($password)) {
-      $ch = curl_init();
-      curl_setopt($ch, CURLOPT_URL,
-        'https://' . $domain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted'
-      );
-      curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
-      curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
-      curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
-      $autoresponders = json_decode(curl_exec($ch), TRUE);
-      curl_close($ch);
-
-      if (!empty($autoresponders)) {
-        foreach ($autoresponders as $autoresponder) {
-          if (!empty($autoresponder['id']) && !empty($autoresponder['title'])) {
-            $autoresponder_list[$autoresponder['id']] = trim($autoresponder['title']);
-          }
+    if (empty($domain) && empty($username) && empty($password)) {
+      return $autoresponder_list;
+    }
+    $full_url = 'https://' . $domain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted';
+    try {
+      $client = \Drupal::httpClient();
+      $request = $client->request('GET', $full_url, [
+        'auth' => [$username, $password],
+      ]);
+      $autoresponders = json_decode($request->getBody(), TRUE);
+      if (empty($autoresponders)) {
+        return $autoresponder_list;
+      }
+      foreach ($autoresponders as $autoresponder) {
+        if (!empty($autoresponder['id']) && !empty($autoresponder['title'])) {
+          $autoresponder_list[$autoresponder['id']] = trim($autoresponder['title']);
         }
       }
     }
+    catch (ClientException $e) {
```

Comment:
Try to avoid handling exceptions silently. At least log error to Drupal logger. It will help developers debug errors.